src/xtgeo/cube/_cube_window_attributes.py

```diff
@@ -311,13 +322,19 @@ def _add_to_attribute_map(self, attr_name: str, values: np.ndarray) -> None:
         attr_map = self._upper.copy()
         attr_map.values = np.ma.masked_invalid(values)
 
+        attr_map.values = np.ma.masked_where(
+            self._mask_map_by_traceidcode.values.mask, attr_map.values
```

Comment:
Would it be possible to simplify and do the comparison against the `self.cube.traceidcodes` directly here, and remove the function to create the `_mask_map_by_traceidcode` ? 
Haven't tested, so might not work :) 
```suggestion
            self.cube.traceidcodes == 2, attr_map.values
```

Reply:
Yes, you are right - nice catch 👍